Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/model/FileListItem.java

```diff
@@ -29,7 +30,18 @@ public FileListItem(File file)
 		}
 		else
 		{
-			String fileExtension = mPath.substring(mPath.lastIndexOf('.'));
+			String fileExtension = null;
+
+			// If the file has no extension, this would throw an exception.
+			try
+			{
+				fileExtension = mPath.substring(mPath.lastIndexOf('.'));
+			}
+			catch (Exception ignored)
+			{
+				// We don't care about files without extensions anyway.
+				mType = TYPE_OTHER;
+			}
```

Comment:
You can eliminate the use of the try-catch, and exception throwing entirely if you check if lastIndexOf returns -1 or not.